ocaml/src/tools/llio2.ml

```diff
@@ -21,22 +21,26 @@ open Lwt_bytes2
 
 module ReadBuffer = struct
     type t = { buf : Lwt_bytes.t;
-               mutable pos : int; }
+               mutable pos : int;
+               too_far : int;
+             }
 
     type 'a deserializer = t -> 'a
 
-    let make_buffer buf pos = { buf; pos; }
+    let make_buffer buf pos too_far = { buf; pos; too_far }
 
     let advance_pos buf delta = buf.pos <- buf.pos + delta
```

Comment:
add an assert here that `buf.pos <= buf.max_pos`
(max_pos is more descriptive for me than too_far)